source/opt/folding_rules.cpp

```diff
@@ -124,6 +124,102 @@ Instruction* NonConstInput(IRContext* context, const analysis::Constant* c,
       inst->GetSingleWordInOperand(in_op));
 }
 
+std::vector<uint32_t> ExtractInts(uint64_t val) {
+  std::vector<uint32_t> words;
+  words.push_back(static_cast<uint32_t>(val));
+  words.push_back(static_cast<uint32_t>(val >> 32));
+  return words;
+}
+
+std::vector<uint32_t> GetWordsFromNumericScalarOrVectorConstant(
+    analysis::ConstantManager* const_mgr, const analysis::Constant* c) {
+  std::vector<uint32_t> words;
+  if (const auto* float_constant = c->AsFloatConstant()) {
```

Comment:
Could you extract the contents of each case into their own function?  It would be cleaner.